### PR TITLE
ATB-1494: ES Build Transition

### PR DESCRIPTION
**Pull request by chiarapascucci-gds**
## Proposed changes
This PR merges changes need to switch from using webpack to esbuild for building the application code

### What changed
- removed webpack config file and related dependencies
- added MetaData object to lambda functions to set them to be build with esbuild through `sam build` command
- moved `samconfig.toml` file to the root of the project directory: this is because when we are manually deploying to dev to a feature stack we are using the `sam deploy` command, which uses this config file to pick the configuration parameters needed by the application. ⚠️ When deploying manually make sure NOT to pass the `-t src/infra/main/template.yaml` flag to `sam deploy` command
- updated w/f that deploy application to point to `.aws-sam/build/` folder where build artefact will now be stored. Removed standalone build step as this is now part of the `package` command
- updated readme 
- upticked dependencies to latest sdk versions

### Why did it change
To switch to using SAM default build method and resolve issue with webpack + latest AWS SDK

### Issue tracking
- [ATB-1494](https://govukverify.atlassian.net/browse/ATB-1494)

## Testing
deployed feature stack through manual deployment and manually tested lambda functions
deployed main stack through GH w/f and pipeline completed successfully including feature tests run

## Checklists
- [x] Did not commit any not-required changes to the `src/infra/**/samconfig.toml`
- [x] Tested changes and included test evidence in the PR, if appropriate
- [x] Included all required tags and other properties for any new resources in the SAM template
- [ ] Ensured that any new resources in the SAM Template follow appropriate naming conventions
- [ ] Ensured that naming of new resources is compatible with deploying multiple stacks with custom stack names during development
- [ ] Ensured that no log lines include PII or other sensitive data
- [ ] Implemented unit testing for any new logic implemented, if appropriate
- [x] Ensured that all commits in this PR are signed
- [ ] Ensured appropriate code coverage is maintained by unit tests
- [ ] Checked SonarCube and ensured no code smells were added

[ATB-1494]: https://govukverify.atlassian.net/browse/ATB-1494?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by amd-gds — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -19,55 +19,50 @@
     "lint:spec:oas": "spectral lint src/specs/api.yaml",
     "lint:spec": "yarn lint:spec:oas",
     "audit:fix": "yarn-audit-fix",
-    "build": "yarn clean && webpack",
-    "clean": "rimraf dist coverage",
-    "package:sam": "sam build -t src/infra/main/template.yaml",
+    "clean": "rimraf .aws-sam coverage",
+    "build": "sam build -t src/infra/main/template.yaml",
     "package:sam:fix": "run-script-os",
```
Comment (TaherKapasi) [inline package.json:24]: are these scripts actually needed anymore or can we loose package:sam:fix?

File: src/infra/main/template.yaml [hunk 2]
```diff
@@ -169,7 +169,7 @@ Resources:
     Properties:
       FunctionName: !Sub "${AWS::StackName}-InterventionsProcessorFunction"
       Description: "Processes intervention events and updates account's intervention status."
-      CodeUri: ../../../dist/interventions-processor-handler
+      CodeUri: ../../handlers
```
Comment (TaherKapasi) [inline src/infra/main/template.yaml:172]: wondering if this can be configured once in global config then does not have to be done per lambda

**Review by TaherKapasi — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (chiarapascucci-gds) [reply package.json:24]: I have tested it a few times and I cannot see that it is changing anything in the template file.. I will remove it and test again that everything builds fine

Comment (chiarapascucci-gds) [reply package.json]: done. Also removed deps that was used to parse the yaml

File: src/infra/main/template.yaml (hunk 2 above)
Comment (chiarapascucci-gds) [reply src/infra/main/template.yaml]: done - also moved kmsKeyArn as they all point to the same key

**Review by MinaMahdi — APPROVED** [review]

**Review by domantas-gds — APPROVED** [review]

**Pull request merged**